### PR TITLE
migrate payment addons to this repo.

### PR DESCRIPTION
**Pull request by jeffery9**

### CODE REVIEW

**Comment by jeffery9** [comment]
update, @leangjia have signed CLA.

**Comment by leangjia** [comment]
@oca-clabot  I have signed ICLA.
@elicoidal  
@jeffery9

**Comment by elicoidal** [comment]
Hi @leangjia @jeffery9 Thanks for the contribution and sorry for the delay.
Let me introduce to you some basic information about contribution in OCA and I will review other points in more comments inside the code.
A more technical review will be necessary as well (I am not technical myself).
In general, please follow the principles: 
- Contribution guidelines: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md
- Please follow the OCA README template:
  https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst
- We recommend to include a screenshot as well to help new comers (nice to have)
- Please use simple headers in all py files following OCA templates
  https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3
- Please sort all imports in alphabetical order following: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#imports

**Comment by elicoidal** [comment]
Please note as well that there is in this repo another 2 PR (#6 and #11) for wc and alipay connectors in v8.
Please review them (comments are welcome) and try to merge your current work inside the existing PR.
allpay and tenpay connector should be proposed as 2 different PR (we usually have one PR per module to ease the review and feedback).

File: README.md [hunk 1]
```diff
@@ -10,4 +10,9 @@ This l10n-china repository is hosted in Github and might contain any module rela
 * Other specific modules (Link to courrier companies, etc.)
 
 [//]: # (addons)
+payment_alipay
```
Comment (elicoidal) [inline README.md:13]: no need for that

File: payment_alipay/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# payment_alipay
+alipay for odoo
+
+this module still in development, use it in production on your own risk
+
+any feedback should be thankful.
+
+if you want technical support, please contact the author and check the link http://item.taobao.com/item.htm?spm=a1z10.1-c.w4004-10020970078.16.EvfHHC&id=44624969034
```
Comment (elicoidal) [inline payment_alipay/README.md:8]: remove commercial information.
Contributors (individuals) are listed in the README while companies are inserted in **openerp**.py (authors).

File: payment_alipay/README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+
+
+Odoo 8.0 支付宝模块中文使用<br>
```
Comment (elicoidal) [inline payment_alipay/README.md:12]: Please check the PR related on Alipay and propose improvements there.

File: payment_alipay/__init__.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
```
Comment (elicoidal) [inline payment_alipay/__init__.py:1]: - Please use simple headers in all py files following OCA templates
  https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3

File: payment_alipay/__openerp__.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+
+{
+    'name': 'Alipay Payment Acquirer',
+    'category': 'Website',
+    'summary': 'Payment Acquirer: Alipay Implementation',
+    'version': '1.0',
```
Comment (elicoidal) [inline payment_alipay/__openerp__.py:7]: For version numbers please check:
- Contribution guidelines: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md

File: payment_alipay/__openerp__.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    'description': """Alipay Payment Acquirer""",
+    'author': 'Odoo CN Community, Jeffery <jeffery9@gmail.com>',
+    'depends': ['payment'],
+    'data': [
+        'views/alipay.xml',
+        'views/payment_acquirer.xml',
+        'data/alipay.xml',
+    ],
+    'installable': True,
+    'price': 499.99,
```
Comment (elicoidal) [inline payment_alipay/__openerp__.py:17]: remove pricing
Add license key

File: payment_alipay/controllers/main.py [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        mysign = util.build_mysign(prestr, _KEY, 'MD5')
+        if mysign != post.get('sign'):
+            return 'false'
+
+        reference = post.get('out_trade_no')
+        notify_id = post.get('notify_id')
+        seller_id = post.get('seller_id')
+        tx = None
+        if reference:
+            tx_ids = request.registry['payment.transaction'].search(cr, uid, [('reference', '=', reference)], context=context)
```
Comment (elicoidal) [inline payment_alipay/controllers/main.py:38]: All your code must be PEP8 compliant (https://www.python.org/dev/peps/pep-0008/)

**Comment by elicoidal** [comment]
Note as well that your code should contain unit tests

File: payment_allpay/__init__.py [hunk 8]
```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
```
Comment (elicoidal) [inline payment_allpay/__init__.py:1]: Please apply the comments in previous module about copyright, openerp.py and PEP8

File: payment_allpay/README.md [hunk 9]
```diff
@@ -0,0 +1,6 @@
+# payment_allPay
+allpay for odoo
+
+this module still in development, use it in production on your own risk
```
Comment (elicoidal) [inline payment_allpay/README.md:4]: not sure what this module is for

File: payment_tenpay/README.md [hunk 10]
```diff
@@ -0,0 +1,8 @@
+# payment_tenpay
+tenpay for odoo
```
Comment (elicoidal) [inline payment_tenpay/README.md:2]: Please apply my comments on format for this module as well

File: payment_tenpay/views/tenpay.xml [hunk 11]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+    <data noupdate="1">
+
+        <template id="tenpay_acquirer_button">
+                <form t-if="acquirer.tenpay_partner_account" t-att-action="tx_url" method="post" target="_self">
+                    <!--<input type="hidden" name="service" t-att-value="tx_values['service']"/>-->
```
Comment (elicoidal) [inline payment_tenpay/views/tenpay.xml:7]: remove dead code

File: payment_weixin/README.md [hunk 12]
```diff
@@ -0,0 +1,8 @@
+# payment_tenpay
```
Comment (elicoidal) [inline payment_weixin/README.md:1]: weixin.
Anyway please check existing PR and try to merge your code in it

File: payment_weixin/views/weixin.xml [hunk 13]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+    <data noupdate="1">
+
+        <template id="weixin_acquirer_button">
+            <form t-if="acquirer.weixin_appid" t-att-action="tx_url" method="post" target="_self" enctype="text/plain">
+                <!--<input type="hidden" name="appid" t-att-value="tx_values['appid']"/>-->
```
Comment (elicoidal) [inline payment_weixin/views/weixin.xml:7]: remove dead code

**Comment by elicoidal** [comment]
Thanks for the contribution! This is an impressive work for a initial version.
I hope you understand my comments but do not hesitate to contact me if necessary.
Once the initial PR have been fixed, we will make a more technical review and then work on the README content.

**Comment by leangjia** [comment]
@elicoidal  Thanks for your help, I appreciate that.

1.Both of payment_weixin and tenpay belong to Tencent company in local.
2.Allpay and alipay are different. Allpay is a kinda Taiwan payment tool. Alipay belongs to Alibaba company in China continent.

BR & THANK YOU

**Comment by elicoidal** [comment]
@leangjia 
I understand: as said, alipay and wechat have already proposed PR in this repo. Can you check them and propose to merge your improvements in those PR.

**Comment by jeffery9** [comment]
this PR should be break into several pieces.

**Comment by elicoidal** [comment]
I close this PR as it is replaced by #16 and #17

**Pull request closed without merging**